lib/profile.go

```diff
@@ -20,46 +19,82 @@ import (
 // changes. all of this code should be refactored into subroutines of general
 // configuration getters & setters
 type ProfileMethods struct {
-	inst *Instance
+	d dispatcher
 }
 
-// CoreRequestsName implements the Request interface
-func (ProfileMethods) CoreRequestsName() string { return "profile" }
+// Name returns the name of this method group
+func (m ProfileMethods) Name() string {
+	return "profile"
+}
 
-// NewProfileMethods creates a ProfileMethods pointer from either a repo
-// or an rpc.Client
-func NewProfileMethods(inst *Instance) *ProfileMethods {
-	return &ProfileMethods{inst: inst}
+// Attributes defines attributes for each method
+func (m ProfileMethods) Attributes() map[string]AttributeSet {
+	return map[string]AttributeSet{
+		"getprofile":      {denyRPC, ""},
+		"setprofile":      {denyRPC, ""},
+		"setprofilephoto": {denyRPC, ""},
+		"setposterphoto":  {denyRPC, ""},
+	}
 }
 
+type ProfileParams struct{}
+
 // GetProfile get's this node's peer profile
-func (m *ProfileMethods) GetProfile(ctx context.Context, in *bool) (*config.ProfilePod, error) {
-	res := &config.ProfilePod{}
-	var err error
+func (m ProfileMethods) GetProfile(ctx context.Context, p *ProfileParams) (*config.ProfilePod, error) {
+	got, _, err := m.d.Dispatch(ctx, dispatchMethodName(m, "getprofile"), p)
+	if res, ok := got.(*config.ProfilePod); ok {
+		return res, err
+	}
+	return nil, dispatchReturnError(got, err)
+}
 
-	if m.inst.http != nil {
-		err = m.inst.http.CallMethod(ctx, AEProfile, http.MethodGet, nil, res)
-		if err != nil {
-			return nil, err
-		}
-		return res, nil
+// SetProfileParams defines parameters for setting parts of a profile
+// Cannot use this to set private keys, your peername, or peer id
+type SetProfileParams struct {
+	Pro *config.ProfilePod
+}
```

Comment:
If we just use `*config.ProfilePod` as our input params, `Dispatch` will attempt to validate the params when they first pass through. `*config.ProfilePod` does not consider a `ProfilePod` valid if it is missing its private key. However, as an input, we should not have a private key filled in at this point. 

My solution was to wrap the profilepod in a `SetProfileParams` struct, but would like feedback here as there may be other places where we need to abstract for a similar reason.

Reply:
I think this is good. The config validation is not necessarily the same as input parameters being valid.